scapy/contrib/rtps/common_types.py

```diff
@@ -321,11 +314,11 @@ class VendorIdPacket(Packet):
     fields_desc = [
         # ByteField("major", 0),
         # ByteField("minor", 0),
-        EnumField(
-            name="vendor_id",
-            default=b"\x00\x00",
-            enum=_rtps_vendor_ids,
-            fmt="2s"
+        StrFixedLenEnumField(
+            "vendor_id",
+            b"\x00\x00",
+            _rtps_vendor_ids,
+            length=2
```

Comment:
Because they use strings for some reason, whereas `ShortEnumField` is only supposed to handle integers. Display is broken on python 3 but thanks to implicit int<->bytes conversion it sorta works, but this all collapses on Python 2.

Reply:
But based on the values, don't you think this should be `ShortEnumField`? Do you think we could use this PR opportunity to change that?

Reply:
You're right, updated. I wanted to spend as little time as possible on this file since I have no use for it :p